crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -622,12 +625,30 @@ impl SpecializedMeshPipeline for MeshPipeline {
         let vertex_buffer_layout = layout.get_layout(&vertex_attributes)?;
 
         let (label, blend, depth_write_enabled);
-        if key.contains(MeshPipelineKey::TRANSPARENT_MAIN_PASS) {
-            label = "transparent_mesh_pipeline".into();
-            blend = Some(BlendState::ALPHA_BLENDING);
+        let pass = key.intersection(MeshPipelineKey::BLEND_RESERVED_BITS);
+        if pass == MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA {
+            label = "premultiplied_alpha_mesh_pipeline".into();
+            blend = Some(BlendState::PREMULTIPLIED_ALPHA_BLENDING);
+            shader_defs.push("PREMULTIPLY_ALPHA".into());
+            shader_defs.push("BLEND_PREMULTIPLIED_ALPHA".into());
             // For the transparent pass, fragments that are closer will be alpha blended
             // but their depth is not written to the depth buffer
             depth_write_enabled = false;
+        } else if pass == MeshPipelineKey::BLEND_MULTIPLY {
+            label = "multiply_mesh_pipeline".into();
```

Comment:
I _can_ change this, but it will be slightly inconsistent with the other two, that do match 1:1 with the key name, e.g.

- `MeshPipelineKey::PREMULTIPLY_ALPHA` → `premultiply_alpha_mesh_pipeline`
- `MeshPipelineKey::OPAQUE` → `opaque_mesh_pipeline`

Let me know what you prefer

Reply:
Maybe POSTMULTIPLY_ALPHA and postmultiply_alpha_mesh_pipeline?

Reply:
Stick with multiply.